### PR TITLE
Start session for ecommerce indexer when run in cli - v2

### PR DESCRIPTION
**Pull request by das-peter**
This is a follow up to #4635 
Turns out there are other locations that subsequently trigger the pricing system and hence need a session.
So I've moved the session creation code into its own function and call that function in pretty much every other function in `IndexUpdater` to ensure the session is ready before sending output.
I wonder if as of now it maybe even would make sense to but the session bootstrap into `IndexUpdater::log()` as this is the point at which the output is sent that could interfere with the later start of the session.

### CODE REVIEW

**Comment by brusch** [comment]
@das-peter can you please resolve the conflict? Thanks a lot!

File: bundles/EcommerceFrameworkBundle/IndexService/Tool/IndexUpdater.php [hunk 1]
```diff
@@ -161,6 +173,8 @@ public static function processUpdateIndexQueue($tenants = null, $maxRounds = nul
             $tenants = [$tenants];
         }
 
+        self::startSession();
```
Comment (NiklasBr) [inline bundles/EcommerceFrameworkBundle/IndexService/Tool/IndexUpdater.php]: Are duplicate `startSession()`s required?

Comment (das-peter) [reply bundles/EcommerceFrameworkBundle/IndexService/Tool/IndexUpdater.php]: Duplicated starts aren't necessary, but it's necessary to ensure the session is ready to use in every possible entry point which could trigger an output before starting the session.
A duplication of session starts is explicitly covered in the helper method which only starts a session if the code runs in CLI and there no session.

Comment (NiklasBr) [reply bundles/EcommerceFrameworkBundle/IndexService/Tool/IndexUpdater.php]: Oh, I was confused by Github's review interface, didn't see that they where in two different methods. It looked like they where just on different sides of an foreach. Sorry! :D

Comment (das-peter) [reply bundles/EcommerceFrameworkBundle/IndexService/Tool/IndexUpdater.php]: No worries :)

**Pull request merged**